components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -0,0 +1,1330 @@
+       try {
+           sentProj = (Project) iUpdate.saveAndReturnObject(proj);
+           sentDat = (Dataset) iUpdate.saveAndReturnObject(dat);
+           Assert.assertTrue(isAdmin);
+       } catch (ServerError se) {
+           Assert.assertFalse(isAdmin);
+       }
+       /* import an image for the normalUser into the normalUser's default group 
+        * and target it into the created Dataset*/
+       final RString imageName = omero.rtypes.rstring(fakeImageFile.getName());
+       final List<List<RType>> result = iQuery.projection(
+               "SELECT id FROM OriginalFile WHERE name = :name ORDER BY id DESC LIMIT 1",
+               new ParametersI().add("name", imageName));
+       final long previousId = result.isEmpty() ? -1 : ((RLong) result.get(0).get(0)).getValue();
+       try {
+           List<String> path = Collections.singletonList(fakeImageFile.getPath());
+           importFileset(path, path.size(), sentDat);
+           Assert.assertTrue(isAdmin);
+       } catch (ServerError se) {
+           Assert.assertFalse(isAdmin);
+       }
+       OriginalFile remoteFile = (OriginalFile) iQuery.findByQuery(
+               "FROM OriginalFile o WHERE o.id > :id AND o.name = :name",
+               new ParametersI().addId(previousId).add("name", imageName));
+       if (isAdmin) {
+           Assert.assertEquals(remoteFile.getDetails().getOwner().getId().getValue(), normalUser.userId);
+           Assert.assertEquals(remoteFile.getDetails().getGroup().getId().getValue(), normalUser.groupId);
+       }
+       /* link the Project and the Dataset */
+       ProjectDatasetLink link = linkProjectDataset(sentProj, sentDat);
+       Image image = (Image) iQuery.findByQuery(
+               "FROM Image WHERE fileset IN "
+               + "(SELECT fileset FROM FilesetEntry WHERE originalFile.id = :id)",
+               new ParametersI().addId(remoteFile.getId()));
+       IObject datasetImageLink = iQuery.findByQuery(
+               "FROM DatasetImageLink WHERE child.id = :id",
+               new ParametersI().addId(image.getId()));
+       IObject projectDatasetLink = iQuery.findByQuery(
+               "FROM ProjectDatasetLink WHERE id = :id",
+               new ParametersI().addId(link.getId()));
+       /* take care of post-import workflows which do not use sudo */
+       if (!isSudoing) {
+           loginUser(lightAdmin);
+       }
+       /* Now check that the ImporterAs can delete the objects
+        * created on behalf of the user */
+       if (deletePassing) {
+           try {
+               doChange(Requests.delete().target(datasetImageLink).build());
+               doChange(Requests.delete().target(projectDatasetLink).build());
+               doChange(Requests.delete().target(image).build());
+               doChange(Requests.delete().target(sentDat).build());
+               doChange(Requests.delete().target(sentProj).build());
+           } catch (ServerError se) {
+           /* not expected */
```

Comment:
If it's not expected then why catch it?
Also, if we must catch it, maybe at least have an `Assert.fail` here then.